src/pages/DelegatoryValidator/StakeDialog.tsx

```diff
@@ -23,24 +23,41 @@ import {
 } from "../Validators/Components/Staking";
 import ContentBox from "../../components/IndividualPageContent/ContentBox";
 import ContentRowSpaceBetween from "../../components/IndividualPageContent/ContentRowSpaceBetween";
+import {DELEGATION_POOL_ADDRESS, OCTA} from "../../constants";
+import {useGlobalState} from "../../GlobalState";
+import {useWallet} from "@aptos-labs/wallet-adapter-react";
+import {ValidatorData} from "../../api/hooks/useGetValidators";
 
 type StakeDialogProps = {
   handleDialogClose: () => void;
   isDialogOpen: boolean;
   accountResource?: Types.MoveResource | undefined;
+  validator: ValidatorData;
+  stakedAmount: string;
+  setStakedAmount: (arg: string) => void;
+  setTransactionHash: (arg: string) => void;
+  setTransactionSucceeded: (arg: boolean) => void;
+  setTransactionInProcess: (arg: boolean) => void;
```

Comment:
usually if you see this many props, it's a good sign that the code is not modularized enough.